cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -185,7 +185,15 @@ func enforceRequirements(flags *applyPlanFlags, args []string, dryRun bool, upgr
 	// If option was specified in both args and config file, args will overwrite the config file.
 	if len(args) == 1 {
 		newK8sVersion = args[0]
-		cfg.KubernetesVersion = newK8sVersion
+		if upgradeApply {
+			// The `upgrade apply` version always overwrites the KubernetesVersion in the returned cfg with the target
+			// version. While this is not the same for `upgrade plan` where the KubernetesVersion should be the old
+			// one (because the call to getComponentConfigVersionStates requires the currently installed version).
+			// This also makes the KubernetesVersion value returned for `upgrade plan` consistent as that command
+			// allows to not specify a target version in which case KubernetesVersion will always hold the currently
+			// installed one.
+			cfg.KubernetesVersion = newK8sVersion
```

Comment:
Just to double check. After this change, if the user passes to plan 1.19.2, while the cluster version is at 1.19.1, would the table for control plane component upgrade still print .2 as the planned upgrade target?

Reply:
Yes. The tables are generated using the version getter (initialized at the return of this func with the value of `newK8sVersion`).

Reply:
Ok thanks!